mirascope/anthropic/tools.py

```diff
@@ -139,3 +160,118 @@ def from_fn(cls, fn: Callable) -> Type[AnthropicTool]:
     def from_base_type(cls, base_type: Type[BaseTypeT]) -> Type[AnthropicTool]:
         """Constructs a `AnthropicTool` type from a `BaseType` type."""
         return convert_base_type_to_tool(base_type, AnthropicTool)
+
+
+def _process_schema(
+    schema: dict[str, Any],
+    parent_type: Optional[str] = None,
+) -> str:
+    schema_xml = ""
+    if "$defs" in schema:
+        schema_xml += "\n<definitions>\n"
+        for def_name, definition in schema["$defs"].items():
+            schema_xml += f"<definition name='{def_name}'>\n"
+            schema_xml += _process_property(definition, parent_type)
+            schema_xml += "</definition>\n"
+        schema_xml += "</definitions>"
+    if "properties" in schema:
+        schema_xml += "\n<parameters>\n"
+        for prop, definition in schema["properties"].items():
+            schema_xml += "<parameter>\n"
+            schema_xml += f"<name>{prop}</name>\n"
+            schema_xml += _process_property(definition, parent_type)
+            schema_xml += "</parameter>\n"
+        schema_xml += "</parameters>\n"
+    elif "items" in schema:
+        schema_xml += _process_property(schema, parent_type)
+    return schema_xml
+
+
+def _process_property(
```

Comment:
any reason we are using string concat vs using ET subelements?
feels like string concat can be very error prone. perhaps something to do for next iteration

Reply:
this is ultimately to save on tokens to avoid using all of the indentation that the ET subelements functionality includes. Maybe I can do it with no indentation? For now going to leave, will add a TODO for later.